bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -65,47 +58,33 @@ public class AwattarBridgeHandler extends BaseBridgeHandler {
     private static final int DATA_REFRESH_INTERVAL = 60;
 
     private final Logger logger = LoggerFactory.getLogger(AwattarBridgeHandler.class);
-    private final HttpClient httpClient;
+
     private @Nullable ScheduledFuture<?> dataRefresher;
     private Instant lastRefresh = Instant.EPOCH;
 
-    private static final String URLDE = "https://api.awattar.de/v1/marketdata";
-    private static final String URLAT = "https://api.awattar.at/v1/marketdata";
-    private String url;
-
     // This cache stores price data for up to two days
     private @Nullable SortedSet<AwattarPrice> prices;
-    private double vatFactor = 0;
-    private double basePrice = 0;
     private ZoneId zone;
-    private final TimeZoneProvider timeZoneProvider;
+
+    private AwattarApi awattarApi;
 
     public AwattarBridgeHandler(Bridge thing, HttpClient httpClient, TimeZoneProvider timeZoneProvider) {
         super(thing);
-        this.httpClient = httpClient;
-        url = URLDE;
-        this.timeZoneProvider = timeZoneProvider;
         zone = timeZoneProvider.getTimeZone();
+
+        awattarApi = new AwattarApi(httpClient, zone);
     }
 
     @Override
     public void initialize() {
         updateStatus(ThingStatus.UNKNOWN);
         AwattarBridgeConfiguration config = getConfigAs(AwattarBridgeConfiguration.class);
-        vatFactor = 1 + (config.vatPercent / 100);
-        basePrice = config.basePrice;
-        zone = timeZoneProvider.getTimeZone();
-        switch (config.country) {
-            case "DE":
-                url = URLDE;
-                break;
-            case "AT":
-                url = URLAT;
-                break;
-            default:
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "@text/error.unsupported.country");
-                return;
+
+        try {
+            awattarApi.initialize(config);
```

Comment:
I would suggest to create the `AwattarApi` instance here instead of re-using the same instance after a configuration change.